modules/ts/include/opencv2/ts.hpp

```diff
@@ -360,6 +360,9 @@ void max(const Mat& src, double s, Mat& dst);
 
 void compare(const Mat& src1, const Mat& src2, Mat& dst, int cmpop);
 void compare(const Mat& src, double s, Mat& dst, int cmpop);
+
+int countExceed(const Mat& src, double value, int cmpop);
+int countExceed(const Mat& src1, const Mat& src2, double value, int cmpop);
```

Comment:
Functions around the added stuff are reference implementations to test optimizations of OpenCV basic functionality. They do not perform any "checks".

"Check" functions should be created as GoogleTest reporters (with "void" result value).
Also you can see take a look on macros with `EXPECT_` prefix from ts/include.

Reply:
> Functions around the added stuff are reference implementations to test optimizations of OpenCV basic functionality. They do not perform any "checks".

Right they don't perform any checks, but this function only "counts" and doesn't "checks".
I expect the check to be done by `EXPECT_LE` like [here](https://github.com/opencv/opencv_contrib/blob/33b68c31b91aaa3b97e999832be1fc22fc6023ab/modules/ximgproc/test/test_fbs_filter.cpp#L111)

I'm not sure if we are on the same page.
Yes, may be the title of this PR was misleading. If so, I can update the title.
If I'm not understanding your opinion correctly, could you guide me to any implementations I should follow, please ?

Reply:
This is wrong place for check helper functions.
There are reference implementations to test optimizations of OpenCV basic functionality.
Lets do not mess them.

> I should follow

I will add more constructive comment later. Proper infrastructure decision is required here.

Reply:
@tomoaki0705 There is no architecture decision yet. Could you please move code of this implementation into related opencv_contrib PR to unblock testing?

Reply:
I pushed so https://github.com/opencv/opencv_contrib/pull/2581 should be able to build/pass without this PR